discovery/discovery_test.go

```diff
@@ -172,67 +129,61 @@ func TestPoolSize(t *testing.T) {
 	assert.NotZero(t, errorLogsAfter-errorLogsBefore)
 }
 
-func TestDbOrchestratorPoolCacheSize(t *testing.T) {
+func TestDBOrchestratorPoolCacheSize(t *testing.T) {
+	assert := assert.New(t)
 	dbh, dbraw, err := common.TempDB(t)
 	defer dbh.Close()
 	defer dbraw.Close()
 	require := require.New(t)
 	require.Nil(err)
 
-	node, _ := core.NewLivepeerNode(nil, "", nil)
-	node.Database = dbh
-	node.Eth = &eth.StubClient{}
-
-	errorLogsBefore := glog.Stats.Error.Lines()
-	// No orchestrators on eth.Stubclient will result in no registered orchs found on chain
-	emptyPool := NewDBOrchestratorPoolCache(node)
-	errorLogsAfter := glog.Stats.Error.Lines()
-	require.NotNil(emptyPool)
-	assert.Equal(t, 0, emptyPool.Size())
-	assert.NotZero(t, errorLogsAfter-errorLogsBefore)
+	sender := &pm.MockSender{}
+	node := &core.LivepeerNode{
+		Database: dbh,
+		Eth:      &eth.StubClient{},
+		Sender:   sender,
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-	addresses := []string{"https://127.0.0.1:8936", "https://127.0.0.1:8937", "https://127.0.0.1:8938"}
-	orchestrators := StubOrchestrators(addresses)
-	node.Eth = &eth.StubClient{Orchestrators: orchestrators}
-	nonEmptyPool := NewDBOrchestratorPoolCache(node)
-	require.NotNil(nonEmptyPool)
-	assert.Equal(t, len(addresses), emptyPool.Size())
-}
+	sender.On("ValidateTicketParams", mock.Anything).Return(nil).Times(3)
```

Comment:
Why do we need this here? I don't see a call to `GetOrchestrators()`

Reply:
me being overzealous late at night probably 😄

But indeed, no O's means we never iterate through the for loop that makes that call.

Reply:
fixed in e8acbd03